homeassistant/components/unifi/device_tracker.py

```diff
@@ -62,111 +60,122 @@
+
+    if new_tracked:
+        async_add_entities(new_tracked)
+
+
+class UniFiClientTracker(ScannerEntity):
+    """Representation of a network device."""
+
+    def __init__(self, client, controller):
+        """Set up tracked device."""
+        self.client = client
         self.controller = controller
-        self._detection_time = detection_time
-        self._ssid_filter = ssid_filter
-        self._monitored_conditions = monitored_conditions
-        self._clients = {}
 
     async def async_update(self):
-        """Get the clients from the device."""
-        try:
-            await self.controller.clients.update()
-            clients = self.controller.clients.values()
-
-        except aiounifi.LoginRequired:
-            try:
-                with async_timeout.timeout(5):
-                    await self.controller.login()
-            except (asyncio.TimeoutError, aiounifi.AiounifiException):
-                clients = []
-
-        except aiounifi.AiounifiException:
-            clients = []
-
-        # Filter clients to provided SSID list
-        if self._ssid_filter:
-            clients = [
-                client for client in clients
-                if client.essid in self._ssid_filter
-            ]
-
-        self._clients = {
-            client.raw['mac']: client.raw
-            for client in clients
-            if (dt_util.utcnow() - dt_util.utc_from_timestamp(float(
-                client.last_seen))) < self._detection_time
-        }
+        """Synchronize state with controller."""
+        await self.controller.request_update()
 
-    async def async_scan_devices(self):
-        """Scan for devices."""
-        await self.async_update()
-        return self._clients.keys()
+    @property
+    def is_connected(self):
+        """Return true if the device is connected to the network."""
+        detection_time = self.controller.unifi_config[CONF_DETECTION_TIME]
 
-    def get_device_name(self, device):
-        """Return the name (if known) of the device.
+        if (dt_util.utcnow() - dt_util.utc_from_timestamp(float(
+                self.client.last_seen))) < detection_time:
+            return True
+        return False
 
-        If a name has been set in Unifi, then return that, else
-        return the hostname if it has been detected.
-        """
-        client = self._clients.get(device, {})
-        name = client.get('name') or client.get('hostname')
-        _LOGGER.debug("Device mac %s name %s", device, name)
-        return name
+    @property
+    def source_type(self):
+        """Return the source type, eg gps or router, of the device."""
+        return SOURCE_TYPE_ROUTER
+
+    @property
+    def name(self) -> str:
+        """Return the name of the device."""
+        return self.client.name or self.client.hostname
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique identifier for this switch."""
+        return 'unifi-dt-{}'.format(self.client.mac)
```

Comment:
The unique id doesn't need any integration or platform identifying info. The core is aware of those already. The mac address is enough.

Reply:
It does if it can be used by other integrations. That reminds me that I need to add a site identifier in there

Reply:
The entity registry uses platform and integration when checking uniqueness. I don't understand what you mean with:
> It does if it can be used by other integrations.

Reply:
If one have multiple sites a mobile device could move between them

Reply:
I'm not following. I mean that we can remove this part: `unifi-dt-`

Reply:
Sure. But I need to add site variable instead

Reply:
Ah, ok.